app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -0,0 +1,151 @@
+package com.example.drawingactivity
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+class DialogUtils(
+    private val context: Context,
+    private val viewModel: DrawViewModel,
+) {
+
+    fun addLayerDialog() {
+        viewModel.isAddDialogOpen = true
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.add_layer_dialog, null)
+        val dialog = AlertDialog.Builder(context).apply {
+            setTitle(context.getString(R.string.addLayerDialogTitle))
+            setView(view)
+            val errorView = view.findViewById<TextView>(R.id.errorText)
+            val layerNameInput = view.findViewById<EditText>(R.id.layerNameField)
+            while (!viewModel.layerHintFound) {
+                val currentHint = context.getString(R.string.layerHint, viewModel.hintIndex)
+                viewModel.createLayerHint(currentHint)
+            }
+            layerNameInput.hint = context.getString(R.string.layerHint, viewModel.hintIndex)
+            if (viewModel.isAddDialogOpen && viewModel.userNewLayerName != null) {
+                layerNameInput.setText(viewModel.userNewLayerName)
+                errorView.isVisible =
+                    layerNameInput.text.toString().length > DrawViewModel.CHAR_LIMIT
+            }
+            layerNameInput.addTextChangedListener {
+                viewModel.userNewLayerName = it.toString().trim()
+                errorView.isVisible = it.toString().trim().length > DrawViewModel.CHAR_LIMIT
+            }
+            setPositiveButton(context.getString(R.string.addLayerPositiveButton)) { _, _ ->
+                var layerName: String = layerNameInput.text.toString().trim()
```

Comment:
shouldn't this have a check for if this is larger than 16 characters, because if they have 17 characters (not counting trimmed whitespace) you can still just submit that. Also there's no check for if the layer name already exists in the list of names. haveing two layer 5's is a no no.